src/colorizer/Dataset.ts

```diff
@@ -120,6 +131,24 @@ export default class Dataset {
     this.times = source.getBuffer(FeatureDataType.U32);
   }
 
+  private async loadCentroids(): Promise<void> {
+    if (!this.centroidsFile) {
+      return;
+    }
+    const url = this.resolveUrl(this.centroidsFile);
+    const source = await this.arrayLoader.load(url);
+    this.centroids = source.getBuffer(FeatureDataType.U16);
+  }
+
+  private async loadBounds(): Promise<void> {
+    if (!this.boundsFile) {
+      return;
+    }
+    const url = this.resolveUrl(this.boundsFile);
+    const source = await this.arrayLoader.load(url);
+    this.bounds = source.getBuffer(FeatureDataType.U16);
+  }
```

Comment:
Out of scope here, but might be about time to work out some generic way to handle these load functions rather than this boilerplate.

Reply:
Nice idea, it could at least turn into a one line `this.bounds = await this.loadToBuffer(this.boundsFile, FeatureDataType.U16);`

Reply:
Fixed! It's not the perfect one-liner we were hoping for but a lot of the redundant code has been removed.